s3fs/tests/test_s3fs.py

```diff
@@ -863,6 +863,26 @@ def test_errors(s3):
         s3.find("s3://")
 
 
+def test_errors_cause_preservings(monkeypatch, s3):
+    # We translate the error, and preserve the original one
+    with pytest.raises(FileNotFoundError) as exc:
+        s3.rm("unknown")
+
+    assert type(exc.value.__cause__).__name__ == "NoSuchBucket"
+
+    async def head_object(*args, **kwargs):
+        raise NoCredentialsError
+
+    monkeypatch.setattr(type(s3.s3), "head_object", head_object)
+
+    # Since the error is not translate, the __cause__ would
+    # be None
+    with pytest.raises(NoCredentialsError) as exc:
+        s3.info("test/a.txt")
```

Comment:
Does this case print the "During handling of the above exception, another exception occurred:" message that generally worries users? Perhaps we should set `ex.__suppress_context__ = True` in every case.

Reply:
> Does this case print the "During handling of the above exception, another exception occurred:" message that generally worries users?

Though in this case (NoCredentialsFound), the exception won't be in the context of anything will it? Considering that it is the only exception. For the rest (like the ones s3fs translates), setting `__cause__` would implicitly set `__suppress_context__` true.